index.js

```diff
@@ -69,64 +69,76 @@ async function run() {
     configs.repoOptions
   );
 
-  return octokit.paginate(workflowRunsRequest).then(workflowRuns => {
-    const artifactPromises = workflowRuns
-      .filter(workflowRun => {
-        const skipWorkflow =
-          configs.skipTags && taggedCommits.includes(workflowRun.head_sha);
-
-        if (skipWorkflow) {
-          console.log(`Skipping tagged run ${workflowRun.head_sha}`);
-
-          return false;
-        }
-
-        return true;
-      })
-      .map(workflowRun => {
-        const workflowRunArtifactsRequest = octokit.actions.listWorkflowRunArtifacts.endpoint.merge(
-          {
-            ...configs.repoOptions,
-            run_id: workflowRun.id,
+  return octokit
+    .paginate(workflowRunsRequest, ({ data }, done) => {
+      const stopPagination = data.find(
+        workflowRun => moment.utc().diff(workflowRun.created_at, "days") > 90
+      );
+
+      if (stopPagination) {
+        done();
```

Comment:
So this is called if any of the `workflowRuns` are older than 90 days, right? In that case, would we miss the whole iteration where some of them might be within the 90 days? Since `data` is not returned. So wouldn't the correct approach would be to check if all `workflowRuns` are older than 90 days?

Reply:
> To stop paginating early, you can call the done() function passed as 2nd argument to the response map function. Note that you still have to return the value you want to map the response to, otherwise the last response will be mapped to undefined.

It's supposed to answer your concern. What this actually means is that calling `done()` won't "halt" the iteration. It's just the way to inform the underlying code to stop the pagination *after* the current page.

Reply:
Perfect, thanks